node/node.go

```diff
@@ -199,6 +199,16 @@ func New(cfg *Config, version string) (*Node, error) { //nolint:gocyclo,funlen
 	}
 
 	rpcHandler := rpc.New(chain, syncReader, throttledVM, version, log).WithGateway(gatewayClient).WithFeeder(client)
+	if cfg.EthNode == "" {
+		log.Warnw("L1 client not found, cannot serve starknet_getMessage")
+	} else {
+		ethClient, err := l1.NewETHClient(cfg.EthNode)
+		if err != nil {
+			return nil, err
+		}
+		rpcHandler.WithETHClient(ethClient)
+	}
+
```

Comment:
I think this code is somewhat being repeated on line 285, I think there is a better way to reduce the repetition.

Reply:
Yup, good point, I'll combine them